gammapy/utils/metadata.py

```diff
@@ -0,0 +1,75 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Metadata base container for Gammapy."""
+import json
+from typing import Optional, Union
+from astropy.coordinates import Angle, EarthLocation, SkyCoord
+from astropy.time import Time
+from astropy.units import Quantity
+import yaml
+from pydantic import BaseModel, validator
+from gammapy.version import version
+
+__all__ = ["MetaData", "CreatorMetaData"]
+
+
+class MetaData(BaseModel):
+    """Base model for all metadata classes in Gammapy."""
+
+    class Config:
+        """Global config for all metadata."""
+
+        arbitrary_types_allowed = True
+        validate_all = True
+        validate_assignment = True
+
+        # provides a recipe to export arbitrary types to json
+        json_encoders = {
+            Angle: lambda v: f"{v.value} {v.unit}",
+            Quantity: lambda v: f"{v.value} {v.unit}",
+            Time: lambda v: f"{v.iso}",
+            EarthLocation: lambda v: f"lon : {v.lon.value} {v.lon.unit}, "
+            f"lat : {v.lat.value} {v.lat.unit}, "
+            f"height : {v.height.value} {v.height.unit}",
+            SkyCoord: lambda v: f"lon: {v.spherical.lon.value} {v.spherical.lon.unit}, "
+            f"lat: {v.spherical.lat.value} {v.spherical.lat.unit}, "
+            f"frame: {v.frame.name} ",
```

Comment:
Can we export a `map`?

Reply:
of course exporting a `Map` to json won't be possible.